### ISSUE TITLE
Opening and ending tag mismatch

### ISSUE THREAD

**Issue by finoradin** [MediaArea/MediaAreaXml#2]
When generating a MediaTrace report for MOV files I am seeing a bug where what should be a `<data>` tag, a `<block>` tag is used…

![screen shot 2015-08-17 at 3 27 19 pm](https://cloud.githubusercontent.com/assets/684940/9314053/8531a9fc-44f4-11e5-9a53-4f902503b257.png)

**Comment by finoradin** [comment]
Here are two samples

https://gist.github.com/finoradin/03fdd9f39dd260d74fac
https://gist.github.com/finoradin/266305b14694715c2051

**Comment by JeromeMartinez** [comment]
Actually the ending tag is OK (2 lines later).
There is a problem with parts having a value (here: a byte with a value of 0x01) **and** complex details about the meaning of the value (one line per known bit)

@dericed How would you imagine to show such value?
On my side, I am thinking to remove the value of the flag and show only the "bitstream" as I do for pure bitstream (e.g. AVC streams):
                            <block offset="500" name="Flags">1</data>
                                <data offset="503" name="IsInternal">Yes</data>
                            </block>
-->
                            <block offset="500" name="Flags">
                                <data offset="500" name="Reserved" moreinfo="(7 bits)">0</data>
                                <data offset="500" name="IsInternal">Yes</data>
                            </block>

**Comment by finoradin** [comment]
Perhaps my eyes are failing me – yes two lines later is </block>, but why is the opening <block> followed by a </data> ?

**Comment by JeromeMartinez** [comment]
> but why is the opening followed by a ?

This is a bug, /data should not be present.

**Comment by dericed** [comment]
@JeromeMartinez, your proposal looks okay, though you're using two <data> to document the same thing, with one showing interpreted data and one showing uninterpreted data.

```
                            <block offset="500" name="Flags">
                                <data offset="500" name="Reserved" moreinfo="(7 bits)">0</data>
                                <data offset="500" name="IsInternal">Yes</data>
                            </block>
```

Possibly as an alternative we could consider the "Flags" as a <block> and then the raw 7 bits as a <block> within a <block>. It would be helpful to clarify if the value of <data> is intended to be 'interpreted' or not.

**Comment by JeromeMartinez** [comment]
Hotfix: Flags information (e.g. "Isinternal") currently not provided with XML output. In https://github.com/MediaArea/MediaInfoLib/commit/caa349ec8883671eedda3918ba67bc8cb1ec6ff0.
So the output should be valid now.
Ticket still open because I want to provide Flags information soon.

**Comment by finoradin** [comment]
Is this in the package repos or just github? Still seeing the issue with 0.7.76 on Ubuntu

**Comment by JeromeMartinez** [comment]
only GitHub, 0.7.77 is planned for next week.
[There are nightly builds](https://mediaarea.net/download/snapshots/binary/) but not for Ubuntu for the moment (planned soon).

**Comment by dericed** [comment]
should be closed, @finoradin can you verify with 0.7.77 official (not nightly)
